.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+repos:
```

Comment:
Again sanity checking: this file basically does the [pre-commit](https://pre-commit.com/) checks for us, a.k.a automating the trivial checks we run before submitting?

On a high level, this is added to improve development experience?

Reply:
Yes exactly.

Discussed offline, pre-commit currently does mypy checks and blocks commits - which can be fairly painful. Will try pre-push instead.

Reply:
Ty!!